### PR TITLE
feat(protocol): enhance nextTxId logics in  DelegateOwner

### PR DESCRIPTION
**Pull request by dantaik**
This PR modifies DelegateOwner to skip nextTxId verification when call.txId is zero, enabling DAO proposals for parallel L2 transactions.

### CODE REVIEW

File: packages/protocol/script/SendMessageToDelegateOwner.s.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "forge-std/src/Script.sol";
+import "../contracts/L2/DelegateOwner.sol";
+import "../contracts/bridge/IBridge.sol";
+import "../test/common/TestMulticall3.sol";
+
+contract SendMessageToDelegateOwner is Script {
+    address public delegateOwner = 0x904aa0aC002532f1410457484893107757683F53;
```
Comment (adaki2004) [inline packages/protocol/script/SendMessageToDelegateOwner.s.sol:10]: Could you please verify the contract on Taiko ?

File: packages/protocol/script/SendMessageToDelegateOwner.s.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        calls[0].callData =
+            abi.encodeCall(DelegateOwner.setAdmin, (0x4757D97449acA795510b9f3152C6a9019A3545c3));
+
+        calls[1].target = address(delegateOwner);
+        calls[1].allowFailure = false;
+        calls[1].callData =
+            abi.encodeCall(DelegateOwner.setAdmin, (0x55d79345Afc87806B690C9f96c4D7BfE2Bca8268));
+
+        DelegateOwner.Call memory dcall = DelegateOwner.Call({
+            txId: 0,
```
Comment (adaki2004) [inline packages/protocol/script/SendMessageToDelegateOwner.s.sol:33]: Maybe worth to comment/remind future ourselves, this `txId` needs to be matched with the `DelegateOwner`'s `nextTxId` to succeed.
```suggestion
            txId: 0, // Has to match with DelegateOwner's nextTxId
```

File: packages/protocol/script/SendMessageToDelegateOwner.s.sol (hunk 1 above)
Comment (dantaik) [reply packages/protocol/script/SendMessageToDelegateOwner.s.sol:10]: Deployed another one and verified the code. 0x2f46c57046672e71d02ed0f0b71dfa42c7f3c488

**Review by adaki2004 — APPROVED** [review]

File: packages/protocol/.solhintignore [hunk 3]
```diff
@@ -3,5 +3,6 @@ lib/
 contracts/test/TestLibRLPReader.sol
 **/contracts/thirdparty/**/*.sol
 /contracts/automata-attestation/
+test/thirdparty/**
```
Comment (dantaik) [inline packages/protocol/.solhintignore:6]:
```suggestion
test/thirdparty/
```

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**